ncm-chkconfig/src/main/pan/components/chkconfig/schema.pan

```diff
@@ -37,6 +33,23 @@ function chkconfig_allow_combinations = {
     true;
 };
 
+@{Enables (on + startstop&#61;true) all arguments as services.
+  If argument starts with a -, it is disabled (off + startstop&#61;true).
+  Use as e.g.
+    "/software/components" &#61; chkconfig("service1", "-service2");
+}
+function chkconfig = {
+    foreach (idx;arg;ARGV) {
+        m = matches(arg, '^(-)?(.*)$');
+        what = 'on';
+        if (m[1] == '-') {
+            what = 'off';
+        };
+        SELF['chkconfig']['service'][escape(m[2])] = dict(what, '', 'startstop', true);
```

Comment:
Can you operate a level lower in the tree? This will cause an error if any part of /software/components is marked final since the compiler cannot tell that you are only modifying the chkconfig subtree.